eng/Analyzers.props

```diff
@@ -1,6 +1,8 @@
 <Project>
   <PropertyGroup>
     <CodeAnalysisRuleset>$(MSBuildThisFileDirectory)CodeAnalysis.ruleset</CodeAnalysisRuleset>
+    <!-- Disable using analyzers in sourcebuild -->
+    <EnableAnalyzers Condition="'$(DotNetBuildFromSource)' == 'true'">false</EnableAnalyzers>
```

Comment:
This doesn't seem to be related to this change right? perhaps we should separate it on a different PR?

Reply:
This is related, as before src and ref assemblies were going through the manual target that @safern wrote which adds the Analyzers to the csc task. Now they go through the same path as test assemblies and other libs like CoreLib. I moved this out from Tools.props and applied the condition here for sourcebuild.